packages/sync-server/src/sync/CentralSyncManager.ts

```diff
@@ -408,17 +417,21 @@ export class CentralSyncManager {
       //
       // Hence, to fix this, we:
       // 1. When starting updating lookup table, use fixed -1 as the tick and treat them as pending updates (SYNC_TICK_FLAGS.SYNC_LOOKUP_PLACEHOLDER)
-      // 2. After updating lookup table is finished, update all the records with tick = -1 to the latest sync tick
+      // 2. After updating lookup table IS FINISHED AND COMMITED (hence we wrap in a new transaction),
+      // update all the records with tick = -1 to the latest sync tick
       // => That way, sync sessions will never miss any records due to timing issue.
       // Note: We do not need to update pending records when it is the initial build
       // because it uses ticks from the actual tables for updated_at_sync_tick
       if (!isInitialBuildOfLookupTable) {
-        await this.database.wrapInTransaction(async (database: TupaiaDatabase) => {
-          // Wrap inside transaction so that any writes to currentSyncTick
-          // will have to wait until this transaction is committed
-          const { tick: currentTick } = await this.tickTockGlobalClock();
-          await updateSyncLookupPendingRecords(database, currentTick);
-        });
+        await this.models.wrapInTransaction(
+          async (transactingModels: SyncServerModelRegistry) => {
+            // Wrap inside transaction so that any writes to currentSyncTick
+            // will have to wait until this transaction is committed
+            const { tick: currentTick } = await this.tickTockGlobalClock(transactingModels);
```

Comment:
Yeah and other time `transactingModels`.
So I think a jsdoc type comment could help to tell the reader why we are passing in `models` at the declaration of the `tickTockGlobalClock` function, rather than using `this.models`